assets/js/components/user-input/UserInputQuestionnaire.js

```diff
@@ -187,11 +242,14 @@ export default function UserInputQuestionnaire() {
 				</UserInputQuestionWrapper>
 			) }
 
-			{ activeSlug === 'preview' && (
+			{ activeSlug === 'preview' && ( isSavingSettings || isNavigating ) && (
+				<ProgressBar />
+			) }
```

Comment:
We need to display this progress bar not only when we are on the preview step. We need to display it when we are saving a single question as well. Let's move it above, something like this:

```javascript
if ( isSavingSettings || isNavigating ) {
	return (
		<ProgressBar />
	);
}

return (
	<Fragment>
		<ProgressBar
			height={ 0 }
			indeterminate={ false }
			progress={ ( activeSlugIndex + 1 ) / USER_INPUT_QUESTIONS_LIST.length }
		/>

		...
	</Fragment>
);
```

Also, please, make sure the progress bar has the same styles as it had when it was in the preview component.